lib/inferno/dsl/fhir_evaluator/evaluator.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require 'dotenv'
+require 'faraday'
+require 'fhir_models'
+require 'thor'
+require 'erb'
+require 'yaml'
+require 'pathname'
+
+require_relative 'config'
+require_relative 'rule'
+require_relative 'evaluation_context'
+require_relative 'evaluation_result'
+require_relative 'dataset_loader'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      class Evaluator
+        def initialize(ig_path, data_path)
+          validate_args(ig_path, data_path)
+
+          Dir.glob(File.join(__dir__, 'rules', '*.rb')).each do |file|
+            require_relative file
+          end
+
+          # IG Import, rule execution, and result output below will be integrated at phase 2 and 3.
+
+          # @ig = File.join(__dir__, 'ig', ig_path)
+
+          # if data_path
+          #   DatasetLoader.from_path(File.join(__dir__, data_path))
+          # else
+          #   ig.examples
+          # end
+
+          # results = evaluate(data, Config.new)
+          # output_results(results, options[:output])
+        end
+
+        def evaluate(data, config = Config.new)
+          context = EvaluationContext.new(@ig, data, config)
+
+          active_rules = []
+          config.data['Rule'].each do |rulename, rule_details|
+            active_rules << rulename if rule_details['Enabled']
+          end
+
+          Rule.descendants.each do |rule|
+            rule.new.check(context) if active_rules.include?(rule.name.demodulize)
+          end
+
+          context.results
+        end
+
+        def validate_args(ig_path, data_path)
```

Comment:
I think we had a miscommunication somewhere - everything related to file paths and reading files should be in the CLI, in `apps/cli/evaluator.rb` . Apologies the file names are the same but I'm not sure what better unique names might be.

So pretty much everything in this file now except the evaluate method should be migrated over there. I think this file was good as of this earlier revision: https://github.com/inferno-framework/inferno-core/blob/5d1ef09fc3cfe428eb52ef0b5be0ec2f90e15f33/lib/inferno/dsl/fhir_evaluator/evaluator.rb

Reply:
Sure I'm fine with changing names to make them more consistent with the rest of the system. Since the validator is `module FHIRResourceValidation` , `class Validator` , we would probably want this module to be `FHIREvaluation` and the DSL class to remain as `Evaluator`. If it helps distinguish things, maybe we want the cli class to be `apps/cli/evaluate.rb` instead of `apps/cli/evalutator.rb`?

Reply:
Agreed. `evaluate` is more aligned with others e.g. `execute`.